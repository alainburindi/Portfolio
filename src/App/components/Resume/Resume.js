import React, { Component } from 'react';
import { connect } from 'react-redux';
import propTypes from 'prop-types';
import profileActions from '../../../Redux/Actions/profile';

const { getProfile } = profileActions;
export class Resume extends Component {
  constructor(props) {
    super(props);
    this.state = {
      firstName: 'Alain',
      lastName: 'Burindi',
      email: 'alainburindi@gmail.com'
    };
  }

  componentDidMount() {
    const { getProfile: getProfileData } = this.props;
    getProfileData();
  }

  render() {
    const { data } = this.props;
    const { email, firstName, lastName } = this.state;
    return (
      <div className="container" id="resume">
        <div className="colums">
          <div className="column is-6 is-offset-3">
            <h1 className="title is-1 has-text-centered">My Resume</h1>
            <hr />
            <div className="content">
              <h3 className="title is-3">Personal details</h3>
              <hr />

              <ul>
                <li>
                  <span className="has-text-weight-bold">First Name : </span>
                  {firstName}
                </li>
                <li>
                  <span className="has-text-weight-bold">Last Name : </span>
                  {lastName}
                </li>
                <li>
                  <span className="has-text-weight-bold">Email : </span>
                  {email}
                </li>
              </ul>
            </div>

            <div className="content">
              <h3 className="title is-3">Github</h3>

              <hr />
              {data ? (
                <ul>
                  <li>
                    <span className="has-text-weight-bold">Username : </span>
                    {data.login}
                  </li>
                  <li>
                    <span className="has-text-weight-bold">Location : </span>
                    {data.location}
                  </li>
                  <li>
                    <span className="has-text-weight-bold">Followers : </span>
                    {data.followers}
                  </li>
                  <li>
                    <span className="has-text-weight-bold">Following : </span>
                    {data.following}
                  </li>
                  <li>
                    <span className="has-text-weight-bold">Repositories : </span>
                    {data.public_repos}
                  </li>
                </ul>
              ) : (
                <i className="fas fa-spinner fa-pulse" />
              )}
            </div>
          </div>
        </div>
      </div>
    );
  }
}

Resume.propTypes = { data: propTypes.object, getProfile: propTypes.func.isRequired };
export const mapStateToProps = state => ({ data: state.profile.data });
export default connect(mapStateToProps,
  { getProfile })(Resume);
