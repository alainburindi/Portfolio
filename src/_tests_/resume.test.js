import mockAxios from 'axios';
import thunk from 'redux-thunk';
import React from 'react';
import promiseMiddleware from 'redux-promise-middleware';
import configureStore from 'redux-mock-store';
import profileActions from '../Redux/Actions/profile';
import { shallow } from '../enzyme';
import profileReducer from '../Redux/Reducers/profile';
import { Resume, mapStateToProps } from '../App/components/Resume/Resume';

const { getProfile } = profileActions;

const middlewares = [thunk, promiseMiddleware];
const mockStore = configureStore(middlewares);
let wrapper;
const getState = {}; // initial state of the store
const store = mockStore(getState);
const props = { getProfile: jest.fn() };

test('should render the resume page', () => {
  mockAxios.get = jest.fn(() => Promise.resolve({ data: { login: 'AlainBurindi' } }));
  wrapper = shallow(<Resume store={store} {...props} />);
  expect(wrapper).toMatchSnapshot();
});

test('should update the props', () => {
  wrapper = shallow(<Resume store={store} data={{ login: 'AlainBurindi' }} getProfile={jest.fn()} />);
  expect(wrapper).toMatchSnapshot();
  expect(props.getProfile).toHaveBeenCalled();
});

test('map state to props', () => {
  mapStateToProps({ profile: { data: { login: 'Alain', follower: 1 } } });
});

test('Action', () => {
  store.dispatch(getProfile());
  mockAxios.get = jest.fn(() => Promise.reject({ data: { message: 'Forbiden' } }));
  store.dispatch(getProfile());
});

test('should have stored all actions', () => {
  expect(store.getActions()[0]).toEqual({
    type: 'GET_PROFILE',
    payload: { login: 'AlainBurindi' }
  });
  expect(store.getActions()[1]).toEqual({ type: 'GET_PROFILE_FAILED' });
});

describe('Reducer', () => {
  test('GET_PROFILE', () => {
    const state = profileReducer({}, { type: 'GET_PROFILE', payload: { login: 'alainburindi' } });
    expect(state).toEqual({ data: { login: 'alainburindi' }, profileFailed: false });
  });
  test('GET_PROFILE_FAILED', () => {
    const state = profileReducer({}, { type: 'GET_PROFILE_FAILED' });
    expect(state).toEqual({
      data: {
        login: 'alainburindi',
        avatar_url: 'https://avatars1.githubusercontent.com/u/41288911?v=4',
        url: 'https://api.github.com/users/alainburindi',
        type: 'User',
        name: 'Alain Burindi',
        location: 'Rwanda',
        public_repos: 16,
        public_gists: 1,
        followers: 1,
        following: 1
      },
      profileFailed: true
    });
  });
  test('DEFAULT', () => {
    const state = profileReducer(undefined, { type: 'UNKNOWN' });
    expect(state).toEqual({});
  });
});
