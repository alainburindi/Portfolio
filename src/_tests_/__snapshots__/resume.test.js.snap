// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the resume page 1`] = `ShallowWrapper {}`;

exports[`should update the props 1`] = `ShallowWrapper {}`;
